addons/report/models/res_company.py

```diff
@@ -22,9 +22,12 @@ def set_report_template(self):
         self.ensure_one()
         if self.env.context.get('report_template', False):
             self.external_report_layout = self.env.context['report_template']
-        if self.env.context.get('default_report_name'):
-            document = self.env.get(self.env.context['active_model']).browse(self.env.context['active_id'])
-            return self.env['report'].get_action(document, self.env.context['default_report_name'], config=False)
+        report = self.env.context.get('default_report_name')
+        if report:
+            return {
+                'type': 'ir.actions.report.xml',
+                'report_name': report,
+            }
```

Comment:
You will get the same behavior only after configuring the company data..This code is for close the wizard automatically.., after you choose the layout.

https://drive.google.com/file/d/0B_qdBJ5LLS_7bEp0QlRmTTlyeWM/view